File: application/core/src/main/java/br/com/beblue/bluerecords/core/usecase/VendaUseCase.java
package br.com.beblue.bluerecords.core.usecase;

import br.com.beblue.bluerecords.core.command.ConsultaVendaCommand;
import br.com.beblue.bluerecords.core.command.RegistraVendaCommand;
import br.com.beblue.bluerecords.core.entitidade.Venda;
import br.com.beblue.bluerecords.core.entitidade.paginacao.Paginacao;

public interface VendaUseCase {

    Venda registrarVenda(RegistraVendaCommand command);

    Paginacao<Venda> consultarVendas(ConsultaVendaCommand command);

    Venda consultarVendaPorId(Integer id);

}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/usecase/venda/VendaUseCase.kt
package br.com.beblue.bluerecords.core.usecase.venda

import br.com.beblue.bluerecords.core.entity.Venda
import br.com.beblue.bluerecords.core.usecase.venda.command.ConsultaVendasCommand
import br.com.beblue.bluerecords.core.usecase.venda.command.RegistrarVendaCommand

interface VendaUseCase {
    fun registrarVenda(command: RegistrarVendaCommand): Venda
    fun consultarVendas(command: ConsultaVendasCommand): List<Venda>
    fun consultarVendaPorId(id: Int): Venda
}